src/components/language-context.js

```diff
@@ -94,15 +94,31 @@ export function useLanguage() {
   return createLanguageAndLocale(useContext(languageContext));
 }
 
+export function useDateFormat() {
+  const { locale } = useLanguage();
+
+  const dateLocale = locale === 'de-DE' ? deDeLocale : enUsLocale;
+  const localePattern = 'P p';
+  const formatDate = (date, pattern) => date
+    ? format(parseISO(date), pattern || localePattern, { locale: dateLocale })
+    : '';
+
+  return {
+    formatDate
+  };
```

Comment:
Lines 100-108 should be memoized (we will only have two cases), otherwise we create a new `formatDate` function and return object of the hook (which in turn will cause unnecessary re-rendering) on each call of the hook (= on each rerender)

Reply:
Done